### ISSUE TITLE
Support GHC 9.0

### ISSUE THREAD

**Issue by expipiplus1** [haskell/actions#14]
It would be great to have a 9.0.* version available.

**Comment by jared-w** [comment]
I posted more about prereleases [here](https://github.com/haskell/actions/issues/15#issuecomment-749830272), but to make a long story short, they're not very consistent in any way right now.

That said, the action has been designed so that "invalid" versions pass through, so if you're up for dealing with cross-platform inconsistencies, something like

```yaml
- uses: haskell/actions/setup
  with:
    ghc-version: '9.0.0.20200925'
```

Should work on Linux and macOS. A newer version of cabal might be required? But I'm not sure. Either way, it'll be somewhat uncharted territory per operating system (especially due to version mismatches across platforms)

**Comment by Bodigrim** [comment]
GHC 9.0 has been released and `ghcup` is able to download it, but `haskell/action` surprisingly fails Ubuntu builds:
```
Installing ghc version 9.0
  Warning: ghc 9.0 was not found in the cache. It will be downloaded.
  If this is unexpected, please check if version 9.0 is pre-installed.
  The list of pre-installed versions is available here: https://help.github.com/en/actions/reference/software-installed-on-github-hosted-runners
  The above list follows a common haskell convention that the three latest major releases of ghc are commonly supported.
  If the list is outdated, please file an issue here: https://github.com/actions/virtual-environments
  by using the appropriate tool request template: https://github.com/actions/virtual-environments/issues/new/choose
  Attempting to install ghc 9.0 using apt-get
  /usr/bin/sudo -- sh -c apt-get -y install ghc-9.0
  Reading package lists...
  Building dependency tree...
  Reading state information...
  E: Unable to locate package ghc-9.0
  E: Couldn't find any package by glob 'ghc-9.0'
  E: Couldn't find any package by regex 'ghc-9.0'
  Attempting to install ghc 9.0 using ghcup
  /opt/hostedtoolcache/ghcup/0.1.12/x64/ghcup install ghc 9.0
  [ Info  ] downloading: https://downloads.haskell.org/~ghc/9.0.1/ghc-9.0.1-x86_64-deb9-linux.tar.xz
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   15  207M   15 31.8M    0     0  36.8M      0  0:00:05 --:--:--  0:00:05 36.8M
  100  207M  100  207M    0     0   128M      0  0:00:01  0:00:01 --:--:--  128M
  [ Info  ] verifying digest of: ghc-9.0.1-x86_64-deb9-linux.tar.xz
  [ Info  ] Unpacking: ghc-9.0.1-x86_64-deb9-linux.tar.xz to /tmp/ghcup-DhLaGg
  [ Info  ] Installing GHC (this may take a while)
  [ Info  ] GHC installation successful
  /opt/hostedtoolcache/ghcup/0.1.12/x64/ghcup set ghc 9.0
  [ Info  ] GHC 9.0.1 successfully set as default version
Error: All install methods for ghc 9.0 failed
```
https://github.com/Bodigrim/tasty-bench/runs/1841211687?check_suite_focus=true#step:3:8

**Comment by jared-w** [comment]
Ah, interesting. This is because of how I check that GHC was actually installed; the action ends up needing to know the full version of GHC or it won't be able to differentiate between, say, `8.8.3` and `8.8.4`. Currently the action does not yet have the knowledge that `9.0` should resolve to `9.0.1`. If you were to use `9.0.1` in the version rather than `9.0`, it should work.

In fact, [it does actually work](https://github.com/Bodigrim/tasty-bench/runs/1841211687?check_suite_focus=true#step:3:35), but `which ghc` is not returning the expected path of `~/.ghcup/ghc/9.0/bin/ghc`, it would be `~/.ghcup/ghc/9.0.1/bin/ghc` (and due to "reasons" I use `ghcup set 9.0.1` to make sure that the ghcup symlink `~/.ghcup/bin/ghc` is the right one despite checking the other path so that I don't have to try and parse the output of `ghc --numeric-version` which is much slower than just checking file paths)

---

tl;dr, use `ghc-version: 9.0.1` and once I put a new release of this out, `9.0` will work as expected :)

**Comment by Bodigrim** [comment]
> once I put a new release of this out, `9.0` will work as expected

How soon could it happen? No rush, but I'm reluctant to put `9.0.1`, because it would require me to update for `9.0.2`, etc. in future.

**Comment by jared-w** [comment]
So I'm in a bit of a quandry right now. Chocolatey does not support GHC 9.0.1 yet, so my only option is to either delay the release or rewire the windows GHC 9.0.1 to use `9.0.1-rc1`.

I ended up taking the latter approach, in the interest of getting things out the door, and I'll update the action when the "real" version of GHC 9.0.1 is available on chocolatey.

**Comment by Bodigrim** [comment]
Thanks!
